instrumentation/aws-sdk/aws-sdk-2.2/testing/src/main/groovy/io/opentelemetry/instrumentation/awssdk/v2_2/AbstractAws2ClientTest.groovy

```diff
@@ -73,6 +80,106 @@ abstract class AbstractAws2ClientTest extends InstrumentationSpecification {
 
   abstract void configureSdkClient(SdkClientBuilder builder)
 
+  def "send DynamoDB #operation request with builder #builder.class.getName() mocked response"() {
+    setup:
+    configureSdkClient(builder)
+    def client = builder
+      .endpointOverride(server.address)
+      .region(Region.AP_NORTHEAST_1)
+      .credentialsProvider(CREDENTIALS_PROVIDER)
+      .build()
+    responseBody.set("")
+    def response = call.call(client)
+
+    if (response instanceof Future) {
+      response = response.get()
+    }
+
+    expect:
+    response != null
+    response.class.simpleName.startsWith(operation)
+    assertDynamoDbRequest(service, operation, path, method, requestId)
+
+    where:
+    [service, operation, method, path, requestId, builder, call] << dynamoDbRequestDataTable()
+  }
+
+  def "send DynamoDB #operation async request with builder #builder.class.getName() mocked response"() {
+    setup:
+    configureSdkClient(builder)
+    def client = builder
+      .endpointOverride(server.address)
+      .region(Region.AP_NORTHEAST_1)
+      .credentialsProvider(CREDENTIALS_PROVIDER)
+      .build()
+    responseBody.set("")
+    def response = call.call(client)
+
+    if (response instanceof Future) {
+      response = response.get()
+    }
+
+    expect:
+    response != null
+    assertDynamoDbRequest(service, operation, path, method, requestId)
+
+    where:
+    [service, operation, method, path, requestId, builder, call] << dynamoDbRequestDataTable()
+  }
+
+  def assertDynamoDbRequest(service, operation, path, method, requestId) {
+    assertTraces(1) {
+      trace(0, 1) {
+        span(0) {
+          operationName "$service.$operation"
+          spanKind CLIENT
+          errored false
+          parent()
+          attributes {
+            "${SemanticAttributes.NET_PEER_NAME.key()}" "localhost"
+            "${SemanticAttributes.NET_PEER_PORT.key()}" server.address.port
+            "${SemanticAttributes.HTTP_URL.key()}" { it.startsWith("${server.address}${path}") }
+            "${SemanticAttributes.HTTP_METHOD.key()}" "$method"
+            "${SemanticAttributes.HTTP_STATUS_CODE.key()}" 200
+            "${SemanticAttributes.HTTP_USER_AGENT.key()}" { it.startsWith("aws-sdk-java/") }
+            "aws.service" "$service"
+            "aws.operation" "${operation}"
+            "aws.agent" "java-aws-sdk"
+            "aws.requestId" "$requestId"
+            "aws.table.name" "sometable"
+            "${SemanticAttributes.DB_SYSTEM.key()}" "dynamodb"
+            "${SemanticAttributes.DB_NAME.key()}" "sometable"
+            "${SemanticAttributes.DB_OPERATION.key()}" "${operation}"
```

Comment:
can we get `db.user`?

Reply:
I don't think so. Client is (usually) configured with AWS access key (id / value) that does not say anything about a particular user.

Reply:
can access key id be used as username? I am not sure how sensitive this info is.

Reply:
In general it can - the AWS AccessKeyId is used in Presigned URLs, API requests etc. Therefore it's usually AFAIK not considered as a secret / sensitive piece of data. @anuraaga any word on this?

Reply:
The access key ID is an ID of the key itself, but not of the IAM role which is closer to what a user is in a MySQL DB. While it could be mapped to a user by resolving the key ID through IAM API or something, I think in general the key ID isn't really useful for tracing and would leave it out.